### PR TITLE
engine: inject synclet if image target has a LiveUpdate [ch2040]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @landism, @nicks,

Please review the following commits I made in branch maiamcc/synclet-inject:

5c147e66573148b9c31eb995ebe36e7d377e6346 (2019-03-28 17:29:48 -0400)
engine: inject synclet if image target has a LiveUpdate [ch2040]

### CODE REVIEW

File: internal/engine/build_and_deployer_live_update_test.go [hunk 1]
```diff
@@ -75,17 +75,18 @@ func runTestCase(t *testing.T, f *bdFixture, tCase testCase) {
...
-
 	if tCase.expectK8sDeploy {
 		expectedYaml := "image: gcr.io/some-project-162817/sancho:tilt-11cd0b38bc3ceb95"
 		if !strings.Contains(f.k8s.Yaml, expectedYaml) {
 			t.Errorf("Expected yaml to contain %q. Actual:\n%s", expectedYaml, f.k8s.Yaml)
 		}
 		assert.Equal(t, tCase.expectSyncletDeploy, strings.Contains(f.k8s.Yaml, sidecar.SyncletImageName), "expected synclet-deploy = %t (deployed yaml was: %s)", tCase.expectSyncletDeploy, f.k8s.Yaml)
+	} else {
+		// Did a container build, expect result to have the container ID we operated on
+		id := manifest.ImageTargetAt(0).ID()
```
Comment (nicks) [inline internal/engine/build_and_deployer_live_update_test.go]: i don't this move is correct? shouldn't we have a result in both cases?

File: internal/engine/build_and_deployer_live_update_test.go [hunk 2]
```diff
@@ -75,17 +75,18 @@ func runTestCase(t *testing.T, f *bdFixture, tCase testCase) {
...
-	assert.Equal(t, k8s.MagicTestContainerID, result.OneAndOnlyContainerID().String())
-
 	if tCase.expectK8sDeploy {
 		expectedYaml := "image: gcr.io/some-project-162817/sancho:tilt-11cd0b38bc3ceb95"
 		if !strings.Contains(f.k8s.Yaml, expectedYaml) {
 			t.Errorf("Expected yaml to contain %q. Actual:\n%s", expectedYaml, f.k8s.Yaml)
 		}
 		assert.Equal(t, tCase.expectSyncletDeploy, strings.Contains(f.k8s.Yaml, sidecar.SyncletImageName), "expected synclet-deploy = %t (deployed yaml was: %s)", tCase.expectSyncletDeploy, f.k8s.Yaml)
+	} else {
+		// Did a container build, expect result to have the container ID we operated on
```
Comment (nicks) [inline internal/engine/build_and_deployer_live_update_test.go]: fwiw, i think DockerCompose builds also return a container ID

File: internal/engine/build_and_deployer_live_update_test.go (hunk 1 above)
Comment (maiamcc) [reply internal/engine/build_and_deployer_live_update_test.go]: ah, i see. Yeah we should get a `result` in all cases, moving that check out of the conditional. (we only expect it to have a container ID if we did a docker compose build or a synclet/localContainerBuild tho.)

File: internal/engine/build_and_deployer_live_update_test.go (hunk 2 above)
Comment (maiamcc) [reply internal/engine/build_and_deployer_live_update_test.go]: yyyeahhh. i'll leave a comment.

Comment (maiamcc) [reply internal/engine/build_and_deployer_live_update_test.go]: hmm actually I'm not sure that's relevant.

**Pull request merged**